hardware_interface/src/resource_manager.cpp

```diff
@@ -51,6 +51,23 @@ class ResourceStorage
   {
   }
 
+  // A new generic method to do initialization checks
+  template <class HardwareT>
+  void check_last_hardware_initialization_and_configure(
+    std::vector<HardwareT> & container, const HardwareInfo & hardware_info)
+  {
+    if (
+      container.back().initialize(hardware_info).id() !=
+      lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+    {
+      throw std::runtime_error(std::string("Failed to initialize '") + hardware_info.name + "'");
+    }
+    if (container.back().configure().id() != lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE)
+    {
+      throw std::runtime_error(std::string("Failed to configure '") + hardware_info.name + "'");
+    }
+  }
```

Comment:
This should work, as @bmagyar suggests.

```suggestion
  void check_hardware_initialization_and_configure(
    HardwareT & hardware, const HardwareInfo & hardware_info)
  {
    if (
      hardware.initialize(hardware_info).id() !=
      lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
    {
      throw std::runtime_error(std::string("Failed to initialize '") + hardware_info.name + "'");
    }
    if (hardware.configure().id() != lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE)
    {
      throw std::runtime_error(std::string("Failed to configure '") + hardware_info.name + "'");
    }
  }
```